### ISSUE TITLE
bug: Modify Request body rule breaks request with ArrayBuffer payload

### ISSUE THREAD

**Issue by lazyvab** [requestly/requestly#265]
### Describe your issue?

If we create a "Modify Request body" rule and use programmatic mode for an XHR request where the original body is of `ArrayBuffer` type, unexpected results are seen.

### Repro steps

Sample JS code in rule:
```js
function modifyRequestBody(args) {
  const { method, url, body } = args;
  // Change request body below depending upon request attributes received in args
  let newBody = body
  console.log("REQUESTLY BODY : ", newBody)
  return newBody;
}
```
Trigger XHR request:
```js
const arrayBufferRequestBody = (new TextEncoder().encode(JSON.stringify({test: "value"}))).buffer;
xhr = new XMLHttpRequest();
xhr.open("POST", "/api");
xhr.send(arrayBufferRequestBody);
```
The request might fail as the payload sent is not in the expected format.

### What Requestly tool were you using?

- [X] Extension
- [ ] Desktop-App
- [ ] Android SDK
- [ ] Selenium

### Your Environment

Chrome

### Requestly Version

23.1.4

**Comment by lazyvab** [comment]
There is a bug in extension currently where anything returned from `modifyRequestBody()` is JSON stringified if it's an object. Whereas in this case, arraybuffers should be sent as it is. 

Meanwhile we roll out the fix, the JS code can be modified to return the value decoded from ArrayBuffer, if the value contained is a JSON.
```js
function modifyRequestBody(args) {
  const { method, url, body } = args;
  // Change request body below depending upon request attributes received in args
  let newBody = body;
  console.log("REQUESTLY BODY : ", newBody);
  return new TextDecoder().decode(newBody);
}
```

**Comment by lazyvab** [comment]
This was fixed in extension `v23.1.4`. Now, the `modifyRequestBody` function will receive the original body as the `body` argument and same will be returned from the function. The body parsed as JSON will be received as the `bodyAsJson` argument.
